File: src/main/java/com/itsci/mju/maebanjumpen/controller/AdminController.java
package com.itsci.mju.maebanjumpen.controller;

import com.itsci.mju.maebanjumpen.dto.AdminDTO;
import com.itsci.mju.maebanjumpen.model.Admin;
import com.itsci.mju.maebanjumpen.service.AdminService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/maeban/admins")
public class AdminController {

    @Autowired
    private AdminService adminService;

    @GetMapping
    public ResponseEntity<List<AdminDTO>> getAllAdmins() {
        List<AdminDTO> admins = adminService.getAllAdmins();
        return ResponseEntity.ok(admins);
    }

    @GetMapping("/{id}")
    public ResponseEntity<AdminDTO> getAdminById(@PathVariable int id) {
        AdminDTO admin = adminService.getAdminById(id);
        return ResponseEntity.ok(admin);
    }

    @PostMapping
    public ResponseEntity<AdminDTO> createAdmin(@RequestBody AdminDTO admin) {
        AdminDTO savedAdmin = adminService.saveAdmin(admin);
        return ResponseEntity.ok(savedAdmin);
    }

    @PutMapping("/{id}")
    public ResponseEntity<AdminDTO> updateAdmin(@PathVariable int id, @RequestBody AdminDTO admin) {
        AdminDTO updatedAdmin = adminService.updateAdmin(id, admin);
        return ResponseEntity.ok(updatedAdmin);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteAdmin(@PathVariable int id) {
        adminService.deleteAdmin(id);
        return ResponseEntity.noContent().build();
    }
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/partyrole/controller/AdminController.kt
package com.itsci.mju.maebanjumpen.partyrole.controller

import com.itsci.mju.maebanjumpen.partyrole.dto.AdminDTO
import com.itsci.mju.maebanjumpen.partyrole.service.AdminService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/maeban/admins")
class AdminController(private val adminService: AdminService) {

    @GetMapping
    fun getAllAdmins(): ResponseEntity<List<AdminDTO>> {
        val admins = adminService.getAllAdmins()
        return ResponseEntity.ok(admins)
    }

    @GetMapping("/{id}")
    fun getAdminById(@PathVariable id: Int): ResponseEntity<AdminDTO> {
        val admin = adminService.getAdminById(id)
        return ResponseEntity.ok(admin)
    }

    @PostMapping
    fun createAdmin(@RequestBody admin: AdminDTO): ResponseEntity<AdminDTO> {
        val savedAdmin = adminService.saveAdmin(admin)
        return ResponseEntity.ok(savedAdmin)
    }

    @PutMapping("/{id}")
    fun updateAdmin(@PathVariable id: Int, @RequestBody admin: AdminDTO): ResponseEntity<AdminDTO> {
        val updatedAdmin = adminService.updateAdmin(id, admin)
        return ResponseEntity.ok(updatedAdmin)
    }

    @DeleteMapping("/{id}")
    fun deleteAdmin(@PathVariable id: Int): ResponseEntity<Void> {
        adminService.deleteAdmin(id)
        return ResponseEntity.noContent().build()
    }
}

